### PR TITLE
Lint: Fix str-format-in-logging, len-as-condition, etc. (3)

### PR DESCRIPTION
**Pull request by terminalmage**
This fixes violations found in salt/modules/ (A-M)

### CODE REVIEW

File: salt/modules/bcache.py [hunk 1]
```diff
@@ -668,7 +662,7 @@ def _bdev(dev=None):
     if not dev:
         return False
     else:
-        return _devbase(os.path.realpath(os.path.join(dev, '../')))
```
Comment (s0undt3ch) [inline salt/modules/bcache.py]: :man_facepalming:

**Review by s0undt3ch — APPROVED** [review]

**Comment by terminalmage** [comment]
I can't reproduce the boto failures, and I can't even see them in Jenkins (403 Forbidden).

**Comment by terminalmage** [comment]
I was finally able to reproduce the boto failures and traced them to an error I made in my lint fixes. Should be resolved now.

**Comment by terminalmage** [comment]
Ported to master in https://github.com/saltstack/salt/pull/56702

**Pull request merged**